hetu-transport/src/main/java/io/hetu/core/transport/execution/buffer/PagesSerdeUtil.java

```diff
@@ -186,22 +210,38 @@ protected Page computeNext()
         private final GenericPagesSerde serde;
         private final InputStream input;
         private final Predicate<InputStream> eof;
+        private final int spillPrefetchReadPages;
+        List<Page> prefetchedPages;
 
         PageReaderDirect(GenericPagesSerde serde, InputStream input, Predicate<InputStream> eof)
+        {
+            this(serde, input, eof, NO_PREFETCH);
+        }
+
+        PageReaderDirect(GenericPagesSerde serde, InputStream input, Predicate<InputStream> eof, int spillPrefetchReadPages)
         {
             this.serde = requireNonNull(serde, "serde is null");
             this.input = requireNonNull(input, "input is null");
             this.eof = requireNonNull(eof, "End of data needs to passed");
+            if (spillPrefetchReadPages > 0) {
+                this.prefetchedPages = new ArrayList<>(spillPrefetchReadPages);
```

Comment:
Use LinkedList instead of ArrayList.. 

and for remove, use removeFirst

Reply:
Fixed